### PR TITLE
Add enable and disable batch actions

### PR DESCRIPTION
**Pull request by LukasHirt**
## Description
We've added batch actions into the accounts listing to provide options to enable and disable accounts.

## Related issues
- Closes https://github.com/owncloud/product/issues/118

## Screenshots
image
image
image
image

### CODE REVIEW

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- ui/store/index.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6113746&bid=20008915)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/tests/acceptance/pageobjects/accountsPage.js  2
         

Clones added
============
- ui/store/index.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6113746&bid=20008915)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/tests/acceptance/pageobjects/accountsPage.js  2
         

Clones added
============
- ui/store/index.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6113746&bid=20008915)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/tests/acceptance/pageobjects/accountsPage.js  2
         

Clones added
============
- ui/store/index.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6113746&bid=20008915)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/tests/acceptance/pageobjects/accountsPage.js  2
         

Clones added
============
- ui/store/index.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6113746&bid=20008915)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/tests/acceptance/pageobjects/accountsPage.js  2
         

Clones added
============
- ui/store/index.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6113746&bid=20008915)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/tests/acceptance/pageobjects/accountsPage.js  2
         

Clones added
============
- ui/store/index.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6113746&bid=20008915)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/tests/acceptance/pageobjects/accountsPage.js  2
         

Clones added
============
- ui/store/index.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6113746&bid=20008915)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/tests/acceptance/pageobjects/accountsPage.js  2
         

Clones added
============
- ui/store/index.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6113746&bid=20008915)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/tests/acceptance/pageobjects/accountsPage.js  2
         

Clones added
============
- ui/store/index.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6113746&bid=20008915)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/tests/acceptance/pageobjects/accountsPage.js  2
         

Clones added
============
- ui/store/index.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6113746&bid=20008915)

File: ui/components/App.vue [hunk 1]
```diff
@@ -1,9 +1,32 @@
...
-        Accounts
-      </h1>
+      <h1 v-text="$gettext('Accounts')" />
+      <oc-grid v-if="selectedAccountsAmount > 0" key="selected-accounts-info" gutter="small" class="uk-flex-middle">
+        <span v-text="selectionInfoText" />
+        <div>
+          <oc-action-drop class="accounts-actions-dropdown">
+            <template v-slot:button>
+              <span class="uk-margin-xsmall-right" v-text="$gettext('Actions')" />
+              <oc-icon name="expand_more" />
```
Comment (kulmann) [inline ui/components/App.vue:11]: Can we flip the icon into the other direction (= use the `expand_less` icon) when the drop is expanded? Only if it's easy to achieve.

File: ui/components/App.vue [hunk 2]
```diff
@@ -1,9 +1,32 @@
...
+            </template>
+            <template v-slot:actions>
+                <oc-button
```
Comment (kulmann) [inline ui/components/App.vue:14]: oc-button needs an `ariaLabel` right?

File: ui/components/App.vue [hunk 3]
```diff
@@ -13,22 +36,72 @@
...
   components: { AccountsList },
   computed: {
     ...mapGetters('Accounts', ['isInitialized', 'getAccountsSorted']),
+    ...mapState('Accounts', ['selectedAccounts']),
+
     accounts () {
       return this.getAccountsSorted
+    },
+
+    selectedAccountsAmount () {
```
Comment (kulmann) [inline ui/components/App.vue:52]: What do you think of `numberOfSelectedAccounts` or `selectedAccountsCount` as var name?

File: ui/components/App.vue [hunk 4]
```diff
@@ -13,22 +36,72 @@
...
+
+    selectionInfoText () {
+      const translated = this.$ngettext('%{ amount } selected user', '%{ amount } selected users', this.selectedAccountsAmount)
+
+      return this.$gettextInterpolate(translated, { amount: this.selectedAccountsAmount })
+    },
+
+    actions () {
+      const actions = []
+      const isAnyAccountDisabled = this.selectedAccounts.some(account => !account.accountEnabled)
```
Comment (kulmann) [inline ui/components/App.vue:64]: If you'd count the number of disabled accounts you would iterate only once and then compare numbers (`number of disabled accounts > 0` and `number of disabled accounts < total number of accounts`).

File: ui/components/accounts/AccountsList.vue [hunk 5]
```diff
@@ -37,6 +46,16 @@ export default {
...
       required: true
     }
+  },
+  computed: {
+    ...mapGetters('Accounts', ['areAllAccountsSelected'])
+  },
+  methods: {
+    ...mapActions('Accounts', ['toggleSelectionAll']),
+
+    toggleAllAccountsSelection () {
```
Comment (kulmann) [inline ui/components/accounts/AccountsList.vue:56]: I don't see a reason for introducing a wrapping function. Could call the `toggleSelectionAll` directly on `@change` in the template. Please clarify if needed.

File: ui/store/index.js [hunk 6]
```diff
@@ -87,6 +107,107 @@ const actions = {
         status: 'danger'
       }, { root: true })
     }
+  },
+
+  toggleSelectionAll ({ commit, getters, state }) {
+    getters.areAllAccountsSelected ? commit('SET_SELECTED_ACCOUNTS', []) : commit('SET_SELECTED_ACCOUNTS', [...state.accounts])
```
Comment (kulmann) [inline ui/store/index.js:113]: You can re-use `RESET_ACCOUNTS_SELECTION` when all accounts are selected

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/tests/acceptance/pageobjects/accountsPage.js  2
         

Clones added
============
- ui/store/index.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6113746&bid=20008915)

File: ui/store/index.js [hunk 7]
```diff
@@ -87,6 +107,107 @@ const actions = {
...
+  },
+
+  async enableAccounts ({ commit, dispatch, state, rootGetters }) {
```
Comment (kulmann) [inline ui/store/index.js:116]: Please refactor the function body of this and `disableAccounts` into one generic function. :-) still good to have dedicated functions for `enableAccounts` and `disableAccounts`, but they should call one generic function doing the heavy lifting.

**Review by kulmann — CHANGES_REQUESTED** [review]
Things I noticed in the UI:
- we should add a `clear selection` link like in the files app
- when you select all users, switch to the files app and come back to the accounts app, the selection of the rows is cleared, but the `select all` checkbox is still checked and the bulk action row (`6 selected users` and the actions dropdown) is still visible. One funny thing about this: if you select one more user (which is possible because the checkboxes are not checked), the bulk action row goes to 7 selected users, while we only have 6, and the `select all` checkbox is unchecked (because the selection doesn't equal the overall number of users anymore)

Also, see comments :-)
Most of the code looks really good already, thank you!

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/tests/acceptance/pageobjects/accountsPage.js  2
         

Clones added
============
- ui/store/index.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6113746&bid=20008915)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/tests/acceptance/pageobjects/accountsPage.js  2
         

Clones added
============
- ui/store/index.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6113746&bid=20008915)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/tests/acceptance/pageobjects/accountsPage.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6113746&bid=20008915)

**Review by kulmann — APPROVED** [review]
🚀

**Pull request merged**